tests/golden/test_mapper_allD_last.qasm

```diff
@@ -1,432 +1,165 @@
-    }
-    cz q[0], q[3]
-    skip 1
-    { # start at cycle 164
-        y90 q[3]
-        ym90 q[0]
-    }
-    cz q[3], q[0]
-    skip 1
-    cz q[3], q[1]
-    skip 1
-    cz q[3], q[6]
-    y90 q[1]
+        cnot q[6], q[4]
+        cnot q[3], q[1]
+    }
+    skip 3
+    swap q[1], q[4]
+    skip 5
+    cnot q[3], q[5]
+    skip 3
+    swap q[3], q[0]
+    skip 3
+    swap q[4], q[6]
+    skip 5
+    cnot q[3], q[1]
+    skip 3
+    swap q[4], q[1]
+    skip 5
+    cnot q[3], q[6]
+    skip 3
+    { # start at cycle 167
+        cnot q[5], q[2]
+        cnot q[0], q[3]
+    }
+    skip 3
     { # start at cycle 171
-        x q[1]
-        ym90 q[3]
-    }
-    cz q[1], q[3]
-    skip 1
-    { # start at cycle 174
-        y90 q[3]
-        ym90 q[1]
-    }
-    cz q[3], q[1]
-    skip 1
-    { # start at cycle 177
-        y90 q[1]
-        ym90 q[3]
-    }
-    cz q[1], q[3]
-    { # start at cycle 179
-        y90 q[0]
-        ym90 q[2]
-    }
-    { # start at cycle 180
-        cz q[0], q[2]
-        cz q[1], q[4]
-    }
-    skip 1
-    { # start at cycle 182
-        y90 q[3]
-        ym90 q[0]
-    }
-    { # start at cycle 183
-        cz q[3], q[0]
-        x q[6]
-    }
-    { # start at cycle 184
-        y90 q[4]
-        y q[6]
-    }
-    { # start at cycle 185
-        y90 q[0]
-        ym90 q[3]
-        cz q[4], q[6]
-    }
-    cz q[0], q[3]
-    { # start at cycle 187
-        y90 q[6]
-        ym90 q[4]
-    }
-    { # start at cycle 188
-        y90 q[3]
-        ym90 q[0]
-        cz q[6], q[4]
-    }
-    cz q[3], q[0]
-    ym90 q[6]
-    cz q[3], q[6]
-    { # start at cycle 192
-        y90 q[4]
-        ym90 q[1]
-    }
-    { # start at cycle 193
-        cz q[4], q[6]
-        cz q[3], q[1]
-    }
-    skip 1
-    { # start at cycle 195
-        y90 q[0]
-        y90 q[1]
-        y90 q[2]
-        y90 q[5]
-        y90 q[6]
-    }
-    { # start at cycle 196
-        x q[1]
-        x q[2]
+        cnot q[0], q[2]
+        cnot q[5], q[3]
+    }
+    skip 3
+    { # start at cycle 175
+        swap q[3], q[1]
+        swap q[5], q[2]
+    }
+    skip 9
+    cnot q[3], q[5]
+    skip 3
+    cnot q[3], q[1]
+    skip 3
+    cnot q[0], q[3]
+    skip 3
+    swap q[0], q[3]
+    skip 9
+    cnot q[2], q[0]
+    skip 3
+    { # start at cycle 211
+        swap q[1], q[4]
+        swap q[2], q[0]
+    }
+    skip 9
+    cnot q[3], q[1]
+    skip 3
+    swap q[1], q[3]
+    skip 9
+    cnot q[0], q[3]
+    skip 3
+    cnot q[3], q[6]
+    skip 3
+    cnot q[3], q[5]
+    skip 3
+    cnot q[3], q[1]
+    skip 3
+    swap q[3], q[6]
+    skip 9
+    cnot q[0], q[3]
+    skip 3
+    x q[3]
+    swap q[3], q[1]
+    skip 3
+    x q[5]
+    swap q[5], q[2]
+    skip 4
+    cnot q[0], q[3]
+    skip 3
+    x q[3]
+    swap q[3], q[5]
+    skip 9
+    cnot q[6], q[3]
+    skip 3
+    x q[3]
+    swap q[3], q[0]
+    skip 5
+    cnot q[6], q[4]
+    skip 3
+    cnot q[6], q[3]
+    skip 3
+    { # start at cycle 310
```

Comment:
Really bad, from around 200 to around 300 cycles.
Are different options given?
Must analyze!
Is only difference really that mapper's gates are ordered by criticality?

Reply:
#493